git-open

```diff
@@ -34,18 +36,35 @@ if [[ -z "$giturl" ]]; then
   exit 1
 fi
 
-# Initial case examples: 'git@example.com:user/project', 'https://example.com:8080/scm/user/project.git/'
+# From git-fetch(5), native protocols:
+# ssh://[user@]host.xz[:port]/path/to/repo.git/
+# git://host.xz[:port]/path/to/repo.git/
+# http[s]://host.xz[:port]/path/to/repo.git/
+# ftp[s]://host.xz[:port]/path/to/repo.git/
+# [user@]host.xz:path/to/repo.git/ - scp-like but is an alternative to ssh.
+
 # Trim "/" and ".git" from the end of the url
 giturl=${giturl%/} giturl=${giturl%.git}
 
 # Trim before last '@' and protocol (*://) from beginning
 uri=${giturl##*@} uri=${uri##*://}
 
-# Trims before first ':' or '/' to get path
-urlpath=${uri#*[/:]}
+# If there isn't a protocol, we can assume it's using the scp syntax which uses ':' to seperate the path.
+[[ $giturl =~ :// ]] && pathsep='/' || pathsep=':'
+
+# Seperate the domain and the urlpath on the first {pathsep}. This also removes the gitport from the domain.
+domain=${uri%%[:$pathsep]*} urlpath=${uri#*$pathsep}
+
+# Allow config options to replace the server or the protocol
+openurl="$protocol://$domain"
+
+function replaceConfig() {
+  config=$(git config --get-urlmatch "open.$1" "$openurl")
+  declare -g -- "$1"="${config:-${!1}}"
```

Comment:
On the stock Bash v3.2.57 that comes with macOS, the `-g` flag causes the following error:

```bash
../git-open: line 63: declare: -g: invalid option
declare: usage: declare [-afFirtx] [-p] [name[=value] ...]
```